src/Runner.php

```diff
@@ -178,17 +183,61 @@ public function run($input = null, $output = null, $app = null, $commandFiles =
         return $statusCode;
     }
 
+    public function setLoader(ClassLoader $loader)
+    {
+        return $this->setCommandFilePluginPrefixes($loader->getPrefixesPsr4());
+    }
+
+    public function setCommandFilePluginPrefixes($prefixes)
+    {
+        $this->commandFilePluginPrefixes = $prefixes;
+        return $this;
+    }
+
+    public function setCommandFilePluginPattern($namespacePattern)
+    {
+        $this->namespacePattern = $namespacePattern;
+        return $this;
+    }
+
+    protected function getPluginCommandClasses()
+    {
+        $commandClasses = [];
+        if ((empty($this->commandFilePluginPrefixes)) || empty($this->namespacePattern)) {
+            return [];
+        }
+        $pattern = '#' . $this->namespacePattern . '#';
+        $commandSearchPaths = [];
+        foreach ($this->commandFilePluginPrefixes as $baseNamespace => $paths) {
+            if (preg_match($pattern, $baseNamespace)) {
+                $commandSearchPaths[$baseNamespace] = $paths;
+            }
+        }
+
+        $discovery = new CommandFileDiscovery();
+        $discovery->setSearchLocations(['Commands']);
```

Comment:
@grasmash Would it make things better for your use cases if the search locations could be altered via a method call that specified the locations? Alternately, we could factor this out into a separate class, and you could call it explicitly with the CommandFileDiscovery object of your choice.